packages/nx/bin/nx.ts

```diff
@@ -93,3 +107,43 @@ function resolveNx(workspace: WorkspaceTypeAndRoot | null) {
     });
   }
 }
+
+/**
+ * Assumes currently running Nx is global install.
+ * Warns if out of date by 1 major version or more.
+ */
+function warnIfUsingOutdatedGlobalInstall(localNxVersion?: string) {
+  const globalVersion = require('../package.json').version;
+  const isOutdatedGlobalInstall =
+    globalVersion &&
+    ((localNxVersion && major(globalVersion) - major(localNxVersion) < 0) ||
+      major(globalVersion) - major(getLatestVersionOfNx()) < 0);
+
+  // Using a global Nx Install
+  if (isOutdatedGlobalInstall) {
+    const bodyLines = [
+      'For more information, see https://nx.dev/more-concepts/global-nx',
+    ];
+    if (!localNxVersion) {
+      bodyLines.unshift(
+        'Its possible that this is causing Nx to not pick up your workspace properly.'
+      );
+    }
+    output.warn({
+      title: `Outdated Global Installation of Nx Detected.`,
+      bodyLines,
+    });
+  }
+}
+
+function getLocalNxVersion(workspace: WorkspaceTypeAndRoot): string {
+  return readJsonFile(
+    require.resolve('nx/package.json', {
+      paths: getNxRequirePaths(workspace.dir),
+    })
+  ).version;
+}
+
+function getLatestVersionOfNx(): string {
+  return execSync('npm view nx@latest version').toString().trim();
```

Comment:
Also try and use pnpm but handle the error if you cannot find the latest version